pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -470,37 +477,39 @@ func (a *ApplicationBackupController) backupVolumes(backup *stork_api.Applicatio
 			if err != nil {
 				return err
 			}
+			for i := 0; i < len(pvcs); i += backupVolumeBatchCount {
+				batch := pvcs[i:min(i+backupVolumeBatchCount, len(pvcs))]
+				volumeInfos, err := driver.StartBackup(backup, batch)
+				if err != nil {
+					// TODO: If starting backup for a drive fails mark the entire backup
+					// as Cancelling, cancel any other started backups and then mark
+					// it as failed
+					message := fmt.Sprintf("Error starting ApplicationBackup for volumes: %v", err)
+					log.ApplicationBackupLog(backup).Errorf(message)
+					a.recorder.Event(backup,
+						v1.EventTypeWarning,
+						string(stork_api.ApplicationBackupStatusFailed),
+						message)
+					backup.Status.Status = stork_api.ApplicationBackupStatusFailed
+					backup.Status.Stage = stork_api.ApplicationBackupStageFinal
+					backup.Status.Reason = message
+					backup.Status.LastUpdateTimestamp = metav1.Now()
+					err = a.client.Update(context.TODO(), backup)
+					if err != nil {
+						return err
+					}
+					return nil
+				}
 
-			volumeInfos, err := driver.StartBackup(backup, pvcs)
-			if err != nil {
-				// TODO: If starting backup for a drive fails mark the entire backup
-				// as Cancelling, cancel any other started backups and then mark
-				// it as failed
-				message := fmt.Sprintf("Error starting ApplicationBackup for volumes: %v", err)
-				log.ApplicationBackupLog(backup).Errorf(message)
-				a.recorder.Event(backup,
-					v1.EventTypeWarning,
-					string(stork_api.ApplicationBackupStatusFailed),
-					message)
-				backup.Status.Status = stork_api.ApplicationBackupStatusFailed
-				backup.Status.Stage = stork_api.ApplicationBackupStageFinal
-				backup.Status.Reason = message
+				backup.Status.Volumes = append(backup.Status.Volumes, volumeInfos...)
+				backup.Status.Status = stork_api.ApplicationBackupStatusInProgress
+				backup.Status.Reason = "Volume backups are in progress"
 				backup.Status.LastUpdateTimestamp = metav1.Now()
 				err = a.client.Update(context.TODO(), backup)
```

Comment:
Since you are updating it here further updates will fail because of the version conflict. Need to read the new object before updating again

Reply:
Update returns the updated version of the CRD. But if the update happens outside or by some other thread, it will fail.
Added fix for the same.